File: backend/Proclamation.Core/Entities/Chore.cs
namespace Proclamation.Core.Entities;

public class Chore
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Reward { get; set; }
    public int? AssignedToId { get; set; }
    public int CreatedById { get; set; }
    public int FamilyId { get; set; }
    public ChoreStatus Status { get; set; } = ChoreStatus.Available;
    public DateTime? DueDate { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
    
    // Navigation properties
    public User? AssignedTo { get; set; }
    public User CreatedBy { get; set; } = null!;
}

public enum ChoreStatus
{
    Available = 1,
    InProgress = 2,
    PendingApproval = 3,
    Completed = 4,
    Cancelled = 5
}

File: ProclamationApp/src/types/chore.ts
export enum ChoreStatus {
  Available = 1,
  InProgress = 2,
  PendingApproval = 3,
  Completed = 4,
  Cancelled = 5,
}

export interface Chore {
  id: number;
  title: string;
  description: string;
  reward: number;
  assignedToId?: number;
  assignedToName?: string;
  createdById: number;
  createdByName: string;
  familyId: number;
  status: ChoreStatus;
  statusName: string;
  dueDate?: string;
  createdAt: string;
  completedAt?: string;
  claimedAt?: string;
  completionNotes?: string;
}

export interface CreateChoreRequest {
  title: string;
  description: string;
  reward: number;
  dueDate?: string;
}

export interface UpdateChoreStatusRequest {
  status: ChoreStatus;
  completionNotes?: string;
}

